src/Agent.Plugins/PipelineArtifact/FilePathUNCServer.cs

```diff
@@ -0,0 +1,95 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Agent.Sdk;
+using Microsoft.TeamFoundation.Build.WebApi;
+using Microsoft.VisualStudio.Services.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.VisualStudio.Services.Content.Common.Tracing;
+
+namespace Agent.Plugins.PipelineArtifact
+{
+    public class FilePathUNCServer
+    {
+        internal async Task UploadAsync(
+            AgentTaskPluginExecutionContext context,
+            Guid projectId,
+            int buildId,
+            string artifactName,
```

Comment:
>artifactName [](start = 19, length = 12)

This will break if I pass in artifactName as "/foo" right?

Reply:
Or this:
Path.Combine("\\kkarothmain\\", "/foo");

---
In reply to: [316900787](https://github.com/microsoft/azure-pipelines-agent/pull/2385#discussion_r316900787) [](ancestors = 316900787)

Reply:
We have restriction on the artifactName to prevent special characters